python/ray/rllib/agents/agent.py

```diff
@@ -684,7 +685,7 @@ def session_creator():
             preprocessor_pref=config["preprocessor_pref"],
             sample_async=config["sample_async"],
             compress_observations=config["compress_observations"],
-            num_envs=config["num_envs_per_worker"],
+            num_envs=num_envs,
```

Comment:
Can we defer env creation in `remote_worker_env.py` to the first poll() instead of doing this? That's the current standard way to avoid expensive initialization, and a lot of algos actually use local evaluator to compute samples (or when num_workers=0).

Reply:
We can and I did. I was not aware of algos using local evaluator to compute samples.